src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -1165,12 +1161,12 @@ namespace winrt::TerminalApp::implementation
     //   data with it's PasteFromClipboard event.
     // Arguments:
     // - eventArgs: the PasteFromClipboard event sent from the TermControl
-    void TerminalPage::_PasteFromClipboardHandler(const IInspectable& /*sender*/,
-                                                  const PasteFromClipboardEventArgs& eventArgs)
+    winrt::fire_and_forget TerminalPage::_PasteFromClipboardHandler(const IInspectable /*sender*/,
+                                                                    const PasteFromClipboardEventArgs eventArgs)
     {
-        this->Dispatcher().RunAsync(CoreDispatcherPriority::High, [eventArgs]() {
```

Comment:
related? maybe us capturing the `PasteFromClipboardEventArgs&` here has something to do with #4118, and capturing the strong ref instead will help?

Reply:
It's hard to say if it'll fix #4118, but it's worth a shot.

Reply:
wow actually yeah that's totally reasonable